.github/workflows/build.yml

```
@@ -136,7 +144,7 @@ jobs:
           publish_dir: ./docs/_build/html
 
   test:
-    needs: [check_conventions, build]
+    needs: ["mypy", "check_conventions", "build"]
```

Can probably just depend on `build` because `build` depends on `mypy` and `check_conventions`.
```suggestion
    needs: ["build"]
```